app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/DateTimePromptWidget.java

```diff
@@ -0,0 +1,169 @@
+package com.igalia.wolvic.ui.widgets.prompts;
+
+import android.content.Context;
+import android.icu.util.Calendar;
+import android.text.format.DateFormat;
+import android.view.Gravity;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import android.widget.DatePicker;
+import android.widget.FrameLayout;
+import android.widget.TimePicker;
+
+import com.igalia.wolvic.R;
+import com.igalia.wolvic.audio.AudioEngine;
+import com.igalia.wolvic.browser.api.WSession;
+
+import androidx.annotation.NonNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+
+public class DateTimePromptWidget extends PromptWidget {
+    private AudioEngine mAudio;
+    private Button mCancelButton, mClearButton, mOkButton;
+    private WSession.PromptDelegate.DateTimePrompt mPrompt;
+    private String format;
+
+    private static Date parseDate(final SimpleDateFormat formatter, final String value, final boolean defaultToNow) {
```

Comment:
Same comment about booleans in APIs. Let's better use a enum and that way we wouldn't need to add clarification comments in the callers